picoquic/packet.c

```diff
@@ -338,77 +251,122 @@ int picoquic_parse_packet_header(
     } else {
-        /* If this is a short header, it should be possible to retrieve the connection
-         * context. This depends on whether the quic context requires cnx_id or not.
-         */
-         uint8_t cnxid_length = (receiving == 0 && *pcnx != NULL) ? (*pcnx)->path[0]->remote_cnxid.id_len : quic->local_cnxid_length;
-         ph->pc = picoquic_packet_context_application;
-         ph->pl_val = 0; /* No actual payload length in short headers */
-
-         if ((int)length >= 1 + cnxid_length) {
-             /* We can identify the connection by its ID */
-             ph->offset = (size_t)1 + picoquic_parse_connection_id(bytes + 1, cnxid_length, &ph->dest_cnx_id);
-             /* TODO: should consider using combination of CNX ID and ADDR_FROM */
-             if (*pcnx == NULL)
-             {
-                 if (quic->local_cnxid_length > 0) {
-                     *pcnx = picoquic_cnx_by_id(quic, ph->dest_cnx_id);
-                 }
-                 else {
-                     *pcnx = picoquic_cnx_by_net(quic, addr_from);
-                 }
-             }
-         } else {
-             ph->ptype = picoquic_packet_error;
-             ph->offset = length;
-             ph->payload_length = 0;
-         }
-      
-         if (*pcnx != NULL) {
-             ph->epoch = 3;
-             ph->version_index = (*pcnx)->version_index;
-             /* If the connection is identified, decode the short header per version ID */
-             switch (picoquic_supported_versions[ph->version_index].version_header_encoding) {
-             case picoquic_version_header_17:
-                 if ((bytes[0] & 0x40) != 0x40) {
-                     /* Check for QUIC bit failed! */
-                     ph->ptype = picoquic_packet_error;
-                 }
-                 else {
-                     ph->ptype = picoquic_packet_1rtt_protected;
-                 }
-                 ph->has_spin_bit = 1;
-                 ph->spin = (bytes[0] >> 5) & 1;
-                 ph->pn_offset = ph->offset;
-                 ph->pn = 0;
-                 ph->pnmask = 0;
-                 ph->key_phase = ((bytes[0] >> 2) & 1); /* Initialize here so that simple tests with unencrypted headers can work */
-                 break;
-             }
-
-             if (length < ph->offset || ph->ptype == picoquic_packet_error) {
-                 ret = -1;
-                 ph->payload_length = 0;
-             } else {
-                 ph->payload_length = (uint16_t)(length - ph->offset);
-             }
-         } else {
-             /* This may be a packet to a forgotten connection */
-             ph->ptype = picoquic_packet_1rtt_protected;
-             ph->payload_length = (uint16_t)((length > ph->offset)?length - ph->offset:0);
-         }
+        ret = picoquic_parse_short_packet_header(quic, bytes, length, addr_from, ph, pcnx, receiving);
     }
 
```

Comment:
This split makes a lot of sense for readability, but it complicates reviewing the changes...

Reply:
I saw that once I opened the PR. I'll try to separate refactorings from functional changes better in the future.